scripts/skills/perks/perk_rf_sanguinary.nut

```diff
@@ -17,78 +18,35 @@ this.perk_rf_sanguinary <- ::inherit("scripts/skills/skill", {
 		this.m.IsHidden = false;
 	}
 
-	function onAnySkillUsed( _skill, _targetEntity, _properties )
+	function isEnabled()
 	{
-		_properties.FatalityChanceMult *= 1.5;
-	}
+		if (this.m.IsForceEnabled || !this.m.RequiresWeapon)
+			return true;
 
-	function onBeforeAnySkillExecuted( _skill, _targetTile, _targetEntity, _forFree )
-	{
-		this.m.IsHitHandlingComplete = false;
-	}
+		if (this.getContainer().getActor().isDisarmed())
+			return false;
 
-	function onBeforeTargetHit( _skill, _targetEntity, _hitInfo )
-	{
-		if (this.isAttackValid(_skill, this.getContainer().getActor(), _targetEntity)) this.m.WasBleeding = _targetEntity.getSkills().hasSkill("effects.bleeding");
+		local weapon = this.getContainer().getActor().getMainhandItem();
+		return weapon != null && weapon.isWeaponType(::Const.Items.WeaponType.Cleaver);
 	}
 
-	// Handles cases where target is already bleeding before the hit
-	// Note: Cannot use this function to check for bleeding applied by the attack's onUse function if
-	// the bleeding is applied after attackEntity e.g. vanilla cleave skill
-	function onTargetHit( _skill, _targetEntity, _bodyPart, _damageInflictedHitpoints, _damageInflictedArmor )
+	function onOtherActorDeath( _killer, _victim, _skill, _deathTile, _corpseTile, _fatalityType )
 	{
-		if (this.m.IsHitHandlingComplete || !this.isAttackValid(_skill, this.getContainer().getActor(), _targetEntity))
+		if (!this.m.IsSpent || _fatalityType == ::Const.FatalityType.None || _killer == null || _killer.getID() != this.getContainer().getActor().getID() || !_killer.isPlacedOnMap() || !::Tactical.TurnSequenceBar.isActiveEntity(_killer))
 			return;
 
-		if (_targetEntity.isAlive() && this.m.WasBleeding)
-		{
-			local actor = this.getContainer().getActor();
-			if (actor.getMoraleState() < ::Const.MoraleState.Steady && actor.getMoraleState() != ::Const.MoraleState.Fleeing)
-			{
-				actor.setMoraleState(actor.getMoraleState() + 1);
-				this.spawnIcon("perk_rf_sanguinary", actor.getTile());
-			}
-
-			this.m.IsHitHandlingComplete = true;
-		}
-	}
-
-	// Handles cases where Bleeding is applied by the attack e.g. during cleave skill's onUse function
-	// Note: Is unable to handle bleeding applied by "scheduledEvent" attacks
-	function onAnySkillExecuted( _skill, _targetTile, _targetEntity, _forFree )
-	{
-		if (this.m.IsHitHandlingComplete || !this.isAttackValid(_skill, this.getContainer().getActor(), _targetEntity))
+		if (_skill == null || !_skill.isAttack() || (!this.m.RequiresWeapon && !_skill.m.IsWeaponSkill) || !this.isEnabled())
```

Comment:
Even if you fix the issue with the `!` above.

There is still the issue that `IsForceEnabled` is not strong enough to force-enable this effect if the skill is not a weapon skill. In that case we abort (`return;`) before `IsForceEnabled` inside the `this.isEnabled()` can do its work.

What exactly is `IsForceEnabled` supposed to do (see remark at the top about comment in member declaration), if not circumventing the weapon restriction?